src/inference/smc.js

```diff
@@ -71,6 +79,20 @@ module.exports = function(env) {
     return this.runCurrentParticle();
   };
 
+  // Error function for error handling
+  // this.throwOnError is true: directly throw error
+  // this.throwOnError is false: return error (string) as infer result
+  SMC.prototype.error = function(errType) {
+    if (this.throwOnError) {
+      throw new Error(errType);
+    } else {
+      this.err = errType;
+      // the way to return differs from enumerate and rejection because of CPS
+      // directly calling this.k doesn't work
+      return this.finish();
```

Comment:
I tested another time and `this.k(this.s, errType)` is working fine. I guess I might have be mistaken.